src/btree/bt_curnext.c

```diff
@@ -449,6 +450,13 @@ __cursor_row_next(
             key->data = WT_INSERT_KEY(ins);
             key->size = WT_INSERT_KEY_SIZE(ins);
 
+            if (WT_IS_HS(session->dhandle) && F_ISSET(&cbt->iface, WT_CURSTD_BOUND_LOWER)) {
```

Comment:
There is already a function to check the bounds below. `__wt_btcur_bounds_early_exit`, If that function doesn't satisfy the requirement for HS, how about changing it to support HS also instead of adding separate code here?